base/headers.go

```diff
@@ -12,6 +12,27 @@ import "strings"
 // Whitespace recognised by SIP protocol.
 const c_ABNF_WS = " \t"
 
+// Maybestring contains a string, or nil.
+type MaybeString interface {
+	implementsMaybeString()
+}
+
+// NoString represents the absence of a string.
+type NoString struct{}
+
+func (n NoString) implementsMaybeString() {}
+
+// String represents an actual string.
+type String struct {
+	S string
```

Comment:
No, you're right, your way is cleaner. I can't really think of any practical consequences of having S be public, and I'd rather avoid the indirection of base.NewString. Comment retracted :)